src/ui/public/state_management/__tests__/monitor_changes.js

```diff
@@ -0,0 +1,107 @@
+import expect from 'expect.js';
+import sinon from 'sinon';
+import monitorStateChanges from 'ui/state_management/monitor_changes';
+import SimpleEmitter from 'ui/utils/SimpleEmitter';
+
+describe('monitorStateChanges', function () {
+  const noop = () => {};
+  const eventTypes = [
+    'save_with_changes',
+    'reset_with_changes',
+    'fetch_with_changes',
+  ];
+  let mockState;
+
+  beforeEach(() => {
+    mockState = new SimpleEmitter();
+  });
+
+  describe('sanity checks', function () {
```

Comment:
I've never expected the `describe` suites to match up with the methods, it's just a way to organize the tests. Here, I'm checking that the argument enforcement works as expected, and that the method throws if you give it the wrong stuff. I could drop the grouping if the output really bothers you though, then you'd get:

`monitorStateChanges should throw if not given a handler function`
`monitorStateChanges should throw if not given a valid cleanup function`